ee/zentral/contrib/intune/models.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from django.db import models
+from django.db.models import F
+from django.urls import reverse
+
+from zentral.core.secret_engines import decrypt_str, encrypt_str, rewrap
+
+
+logger = logging.getLogger("zentral.contrib.intune.models")
+
+
+class Tenant(models.Model):
+    business_unit = models.ForeignKey("inventory.BusinessUnit", on_delete=models.PROTECT)
+    name = models.TextField(unique=True)
+    description = models.TextField(blank=True, null=True)
+    # Authentication
+    tenant_id = models.TextField(help_text="The microsoft Azure Tenant ID")
```

Comment:
See if we could use a UUID field here, to restrict a bit the format. Same about the other fields. If not then CharFields probably, and you could remove the override for the form widgets.